ksqldb-serde/src/main/java/io/confluent/ksql/serde/KeyFormat.java

```diff
@@ -97,6 +106,30 @@ public String getFormat() {
     return format.getProperties();
   }
 
+  @JsonIgnore
+  public boolean supportKeyTypes(final List<SqlType> sqlTypes) {
```

Comment:
Rather than having this logic here, I think it'd be better to have it in the `Format` interface itself. We can add a new method `boolean supportsType(final SqlType sqlType)` and have each `Format` implementation define which types are supported by the particular format, and avoid this switch statement entirely.

Reply:
I think the tricky scenario we hit here is that we are using `FormatInfo`, which contains a format string, not the actual `Format` type. I'm afraid we couldn't get rid of the switch condition, at a minimum we need a switch for converting string to `Format`

Reply:
`FormatFactory#of()` addresses this exact use case: https://github.com/confluentinc/ksql/blob/3bf8531d0f67de2590ed8f945033b58f68e38a0f/ksqldb-serde/src/main/java/io/confluent/ksql/serde/FormatFactory.java#L48